kiwi_keg/image_definition.py

```diff
@@ -78,36 +78,41 @@ def populate(self) -> None:
                 'Error parsing image data: {error}'.format(error=issue)
             )
 
-        include_paths = self._data.get('include-paths')
         # load profile sections
         if 'profiles' in self._data:
-            for profile_name, profile_data in self._data['profiles'].items():
-                profile: Dict = {}
-                for item, value in profile_data.items():
-                    if item == 'include':
-                        for inc in value:
-                            utils.rmerge(
-                                utils.get_yaml_tree(
-                                    inc,
-                                    self._data_roots,
-                                    include_paths
-                                ),
-                                profile
-                            )
-                    else:
-                        utils.rmerge({item: value}, profile_data)
-                self._data['profiles'][profile_name].update(profile)
+            self.update_profiles(self._data.get('include-paths'))
 
         # expand unprofiled contents section (for single build)
         if 'contents' in self._data:
-            contents: Dict = {}
-            for inc in self._data['contents'].get('include'):
-                utils.rmerge(
-                    utils.get_yaml_tree(
-                        inc,
-                        self._data_roots,
-                        include_paths
-                    ),
-                    contents
-                )
-            self._data['contents'].update(contents)
+            self.update_contents(self._data.get('include-paths'))
+
+    def update_profiles(self, include_paths):
+        for profile_name, profile_data in self._data['profiles'].items():
```

Comment:
This will fail with a stack trace if no 'profiles' key exists. I'd say it should only run when ```if 'profiles' in self.data```

Also I suggest to use the property (data) and not a direct read on _data

Reply:
> This will fail with a stack trace if no 'profiles' key exists. I'd say it should only run when `if 'profiles' in self.data`
>
Done.  

> Also I suggest to use the property (data) and not a direct read on _data
>
Good catch, using the properties now, added two more and a setter as well.